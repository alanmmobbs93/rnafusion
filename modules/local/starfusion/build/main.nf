process STARFUSION_BUILD {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container 'community.wave.seqera.io/library/dfam_hmmer_minimap2_samtools_pruned:63e3d21ca68ea531'

    input:
    tuple val(meta), path(fasta)
    tuple val(meta2), path(gtf)
    path fusion_annot_lib
    val dfam_species

    output:
    path "ctat_genome_lib_build_dir"  , emit: reference

    script:
    if (dfam_species != "human" && dfam_species != "mouse") {
        error "Invalid species for --dfam_db. Only 'human' or 'mouse' are accepted. Provided: ${dfam_species}"
    }
    def args = task.ext.args ?: ''
    """
    prep_genome_lib.pl \\
        --genome_fa $fasta \\
        --gtf $gtf \\
        --dfam_db ${dfam_species} \\
        --pfam_db current \\
        --fusion_annot_lib $fusion_annot_lib \\
        ${args} \\
        --CPU $task.cpus

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        STAR-Fusion: \$(STAR-Fusion --version 2>&1 | grep -i 'version' | sed 's/STAR-Fusion version: //')
    END_VERSIONS
    """

    stub:
    """
    mkdir ctat_genome_lib_build_dir
    touch ctat_genome_lib_build_dir/AnnotFilterRule.pm
    gzip -c /dev/null > ctat_genome_lib_build_dir/PFAM.domtblout.dat.gz
    touch ctat_genome_lib_build_dir/ref_annot.gtf.gene_spans
    touch ctat_genome_lib_build_dir/ref_genome.fa.mm2
    touch ctat_genome_lib_build_dir/ref_genome.fa.ntf
    gzip -c /dev/null > ctat_genome_lib_build_dir/blast_pairs.dat.gz
    touch ctat_genome_lib_build_dir/ref_annot.cdna.fa
    touch ctat_genome_lib_build_dir/ref_annot.gtf.mini.sortu
    touch ctat_genome_lib_build_dir/ref_genome.fa.ndb
    touch ctat_genome_lib_build_dir/ref_genome.fa.nto
    touch ctat_genome_lib_build_dir/blast_pairs.idx
    touch ctat_genome_lib_build_dir/ref_annot.cdna.fa.idx
    touch ctat_genome_lib_build_dir/ref_annot.gtf.mm2.splice.bed
    touch ctat_genome_lib_build_dir/ref_genome.fa.nhr
    touch ctat_genome_lib_build_dir/ref_genome.fa.star.idx
    touch ctat_genome_lib_build_dir/__chkpts
    touch ctat_genome_lib_build_dir/ref_annot.cds
    touch ctat_genome_lib_build_dir/ref_annot.pep
    touch ctat_genome_lib_build_dir/ref_genome.fa.nin
    touch ctat_genome_lib_build_dir/trans.blast.align_coords.align_coords.dat
    gzip -c /dev/null > ctat_genome_lib_build_dir/fusion_annot_lib.gz
    touch ctat_genome_lib_build_dir/ref_annot.cdsplus.fa
    touch ctat_genome_lib_build_dir/ref_annot.prot_info.dbm
    touch ctat_genome_lib_build_dir/ref_genome.fa.njs
    touch ctat_genome_lib_build_dir/trans.blast.align_coords.align_coords.dbm
    touch ctat_genome_lib_build_dir/fusion_annot_lib.idx
    touch ctat_genome_lib_build_dir/ref_annot.cdsplus.fa.idx
    touch ctat_genome_lib_build_dir/ref_genome.fa
    touch ctat_genome_lib_build_dir/ref_genome.fa.not
    gzip -c /dev/null > ctat_genome_lib_build_dir/trans.blast.dat.gz
    touch ctat_genome_lib_build_dir/pfam_domains.dbm
    touch ctat_genome_lib_build_dir/ref_annot.gtf
    touch ctat_genome_lib_build_dir/ref_genome.fa.fai
    touch ctat_genome_lib_build_dir/ref_genome.fa.nsq

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        STAR-Fusion: \$(STAR-Fusion --version 2>&1 | grep -i 'version' | sed 's/STAR-Fusion version: //')
    END_VERSIONS
    """

}
