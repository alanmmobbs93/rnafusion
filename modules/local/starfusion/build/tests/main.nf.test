nextflow_process {

    name "Test Process STARFUSION_BUILD"
    script "../main.nf"
    process "STARFUSION_BUILD"

    test("STARFUSION_BUILD - human - chr4") {

        when {
            process {
                """
                input[0] = [
                    [ id:'fasta' ],
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/ensembl/Homo_sapiens.GRCh38.102.chr4.1700000-54900000.fa")
                ]
                input[1] = [
                    [ id:'gtf' ],
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/ensembl/Homo_sapiens.GRCh38.102.chr4.gtf")
                ]
                """
            }
        }

        then {
            assert snapshot(process.out).match()
        }

    }

    test("STARFUSION_BUILD - human - chr4") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'fasta' ],
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/ensembl/Homo_sapiens.GRCh38.102.chr4.1700000-54900000.fa")
                ]
                input[1] = [
                    [ id:'gtf' ],
                    file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/rnafusion/testdata/reference/ensembl/Homo_sapiens.GRCh38.102.chr4.gtf")
                ]
                """
            }
        }

        then {
            assert snapshot(process.out).match()
        }

    }

}
