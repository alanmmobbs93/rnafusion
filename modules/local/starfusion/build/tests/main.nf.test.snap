{
    "STARFUSION_BUILD - human - minigenome": {
        "content": [
            {
                "0": [
                    [
                        "AnnotFilterRule.pm:md5,5391fcc58d9c71cd1f0e45668c5ec597",
                        "PFAM.domtblout.dat.gz:md5,6d8ba244e68072be23895ec22c8a8ce4",
                        [
                            "_fusion_annot_lib.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "_prot_info_db.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "annotfiltrule_cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "blast_pairs.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "cp_gene_blast_pairs.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "cp_pfam_dat.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "cp_ref_annot_cdna.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "fusion_annot_lib.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "index_pfam_hits.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "index_ref_annot_cdna.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "makeblastdb.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "mm2.splice_bed.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "mm2_genome_idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.cdsplus.dfam_masked.fa.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.cdsplus.dfam_masked.fa.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.gtf.gene_spans.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.gtf.mini.sortu.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.gtf.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome_fai.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "trans.blast.dat.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "trans.blast.dat.index.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "validate_ctat_genome_lib.ok:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ],
                        "blast_pairs.dat.gz:md5,06d08c55cfa099ccb56d312ca0970729",
                        "blast_pairs.idx:md5,884cc8c9886409f7ec0fdf8586ffb78d",
                        "fusion_annot_lib.gz:md5,23d82a5da81f91ca4e1ecd6481992a12",
                        "fusion_annot_lib.idx:md5,39ab4cd5eab589c5fce9b4bfed82c729",
                        "pfam_domains.dbm:md5,04c75694f8e36461b43d1c693fdf3414",
                        "ref_annot.cdna.fa:md5,fe801b66cfab298c575971401e93c18c",
                        "ref_annot.cdna.fa.idx:md5,3c19e33d5424174f1a3de1f21f6746ff",
                        "ref_annot.cds:md5,4376fb48bd8bf47f13854b3f6bba8297",
                        "ref_annot.cdsplus.fa:md5,71e98a5b5d8cf371ec4b5db32c19120e",
                        "ref_annot.cdsplus.fa.idx:md5,8cb47913bfb0d73d3f55cd5ceafa8bfe",
                        "ref_annot.gtf:md5,5ce8afe99ef3940a877a04caeacf9181",
                        "ref_annot.gtf.gene_spans:md5,f178dd54a1f81174dfd203e212937e63",
                        "ref_annot.gtf.mini.sortu:md5,1d29ccecdbb7b40a99c84a02d6c2c1be",
                        "ref_annot.gtf.mm2.splice.bed:md5,340585ea1843bf06bf555575ddecf28c",
                        "ref_annot.pep:md5,fe3f58b430d99c06c816521ac439c4df",
                        "ref_annot.prot_info.dbm:md5,6983b8dccd3e3f920566fab403a15090",
                        "ref_genome.fa:md5,ad699c56ed38566c7d3e9579486b1706",
                        "ref_genome.fa.fai:md5,e3f74a27219b33ae80fd5de5cbeaf32b",
                        "ref_genome.fa.mm2:md5,ce50979ea284748eb9f84ae88cfd930e",
                        "ref_genome.fa.ndb:md5,6ea574753b557610f62f6e4ab79e19f5",
                        "ref_genome.fa.nhr:md5,50f28dae71683c4394bfaf94a1ef4392",
                        "ref_genome.fa.nin:md5,167a13425cf4aac970ae936bd43cbd43",
                        "ref_genome.fa.njs:md5,7350b9e036410da25ed2c0d717649221",
                        "ref_genome.fa.not:md5,1e53e9d08f1d23af0299cfa87478a7bb",
                        "ref_genome.fa.nsq:md5,d2361e7871ce4cf51181c112a48f191b",
                        "ref_genome.fa.ntf:md5,de1250813f0c7affc6d12dac9d0fb6bb",
                        "ref_genome.fa.nto:md5,33cdeccccebe80329f1fdbee7f5874cb",
                        [
                            "Genome:md5,9e3efdd0901cabb5a2d589664a63b372",
                            "Log.out:md5,944cceca8093617ab4ece780628459f5",
                            "SA:md5,7dd9083264be9c6a2194d990bc10d237",
                            "SAindex:md5,ac4711df685109e04356db9e9cb9fb7f",
                            "build.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "chrLength.txt:md5,e02cd536b7281b894246863b160d5d06",
                            "chrName.txt:md5,f4d0d6595f423084e6b9472e40dfe6e8",
                            "chrNameLength.txt:md5,07a67d7ac441d7d30d80840b0927e717",
                            "chrStart.txt:md5,e2031239a74fe5ee9051e9364e4f608a",
                            "exonGeTrInfo.tab:md5,3c35618d07a8e35a0f9108699fcdda42",
                            "exonInfo.tab:md5,bcbb3f32fa31fe504cc737f337ad341c",
                            "geneInfo.tab:md5,db5db4b6e003904e9908fce7c05f0125",
                            "genomeParameters.txt:md5,aabd07882af60af4bba0438a475e4e1a",
                            "sjdbInfo.txt:md5,e4cc1bbf8bd687cfc3d7c2c702e6def7",
                            "sjdbList.fromGTF.out.tab:md5,8f3e8604b00d4067e4eb80aa476a8113",
                            "sjdbList.out.tab:md5,5d78dd49d5db24ca2c056b7ebe5c2059",
                            "transcriptInfo.tab:md5,b758c0ccaddcf0453bab5905b3cec4a2"
                        ],
                        "trans.blast.align_coords.align_coords.dat:md5,9f6b7a75aea03a9671190be25ecdd4c2",
                        "trans.blast.align_coords.align_coords.dbm:md5,6000e63ebabe57652cfa8858ef6cc484",
                        "trans.blast.dat.gz:md5,85ba5ea96c566f751ad83a3e4b8ab128"
                    ]
                ],
                "reference": [
                    [
                        "AnnotFilterRule.pm:md5,5391fcc58d9c71cd1f0e45668c5ec597",
                        "PFAM.domtblout.dat.gz:md5,6d8ba244e68072be23895ec22c8a8ce4",
                        [
                            "_fusion_annot_lib.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "_prot_info_db.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "annotfiltrule_cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "blast_pairs.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "cp_gene_blast_pairs.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "cp_pfam_dat.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "cp_ref_annot_cdna.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "fusion_annot_lib.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "index_pfam_hits.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "index_ref_annot_cdna.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "makeblastdb.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "mm2.splice_bed.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "mm2_genome_idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.cdsplus.dfam_masked.fa.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.cdsplus.dfam_masked.fa.idx.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.gtf.gene_spans.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.gtf.mini.sortu.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_annot.gtf.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome.fa.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "ref_genome_fai.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "trans.blast.dat.cp.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "trans.blast.dat.index.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "validate_ctat_genome_lib.ok:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ],
                        "blast_pairs.dat.gz:md5,06d08c55cfa099ccb56d312ca0970729",
                        "blast_pairs.idx:md5,884cc8c9886409f7ec0fdf8586ffb78d",
                        "fusion_annot_lib.gz:md5,23d82a5da81f91ca4e1ecd6481992a12",
                        "fusion_annot_lib.idx:md5,39ab4cd5eab589c5fce9b4bfed82c729",
                        "pfam_domains.dbm:md5,04c75694f8e36461b43d1c693fdf3414",
                        "ref_annot.cdna.fa:md5,fe801b66cfab298c575971401e93c18c",
                        "ref_annot.cdna.fa.idx:md5,3c19e33d5424174f1a3de1f21f6746ff",
                        "ref_annot.cds:md5,4376fb48bd8bf47f13854b3f6bba8297",
                        "ref_annot.cdsplus.fa:md5,71e98a5b5d8cf371ec4b5db32c19120e",
                        "ref_annot.cdsplus.fa.idx:md5,8cb47913bfb0d73d3f55cd5ceafa8bfe",
                        "ref_annot.gtf:md5,5ce8afe99ef3940a877a04caeacf9181",
                        "ref_annot.gtf.gene_spans:md5,f178dd54a1f81174dfd203e212937e63",
                        "ref_annot.gtf.mini.sortu:md5,1d29ccecdbb7b40a99c84a02d6c2c1be",
                        "ref_annot.gtf.mm2.splice.bed:md5,340585ea1843bf06bf555575ddecf28c",
                        "ref_annot.pep:md5,fe3f58b430d99c06c816521ac439c4df",
                        "ref_annot.prot_info.dbm:md5,6983b8dccd3e3f920566fab403a15090",
                        "ref_genome.fa:md5,ad699c56ed38566c7d3e9579486b1706",
                        "ref_genome.fa.fai:md5,e3f74a27219b33ae80fd5de5cbeaf32b",
                        "ref_genome.fa.mm2:md5,ce50979ea284748eb9f84ae88cfd930e",
                        "ref_genome.fa.ndb:md5,6ea574753b557610f62f6e4ab79e19f5",
                        "ref_genome.fa.nhr:md5,50f28dae71683c4394bfaf94a1ef4392",
                        "ref_genome.fa.nin:md5,167a13425cf4aac970ae936bd43cbd43",
                        "ref_genome.fa.njs:md5,7350b9e036410da25ed2c0d717649221",
                        "ref_genome.fa.not:md5,1e53e9d08f1d23af0299cfa87478a7bb",
                        "ref_genome.fa.nsq:md5,d2361e7871ce4cf51181c112a48f191b",
                        "ref_genome.fa.ntf:md5,de1250813f0c7affc6d12dac9d0fb6bb",
                        "ref_genome.fa.nto:md5,33cdeccccebe80329f1fdbee7f5874cb",
                        [
                            "Genome:md5,9e3efdd0901cabb5a2d589664a63b372",
                            "Log.out:md5,944cceca8093617ab4ece780628459f5",
                            "SA:md5,7dd9083264be9c6a2194d990bc10d237",
                            "SAindex:md5,ac4711df685109e04356db9e9cb9fb7f",
                            "build.ok:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "chrLength.txt:md5,e02cd536b7281b894246863b160d5d06",
                            "chrName.txt:md5,f4d0d6595f423084e6b9472e40dfe6e8",
                            "chrNameLength.txt:md5,07a67d7ac441d7d30d80840b0927e717",
                            "chrStart.txt:md5,e2031239a74fe5ee9051e9364e4f608a",
                            "exonGeTrInfo.tab:md5,3c35618d07a8e35a0f9108699fcdda42",
                            "exonInfo.tab:md5,bcbb3f32fa31fe504cc737f337ad341c",
                            "geneInfo.tab:md5,db5db4b6e003904e9908fce7c05f0125",
                            "genomeParameters.txt:md5,aabd07882af60af4bba0438a475e4e1a",
                            "sjdbInfo.txt:md5,e4cc1bbf8bd687cfc3d7c2c702e6def7",
                            "sjdbList.fromGTF.out.tab:md5,8f3e8604b00d4067e4eb80aa476a8113",
                            "sjdbList.out.tab:md5,5d78dd49d5db24ca2c056b7ebe5c2059",
                            "transcriptInfo.tab:md5,b758c0ccaddcf0453bab5905b3cec4a2"
                        ],
                        "trans.blast.align_coords.align_coords.dat:md5,9f6b7a75aea03a9671190be25ecdd4c2",
                        "trans.blast.align_coords.align_coords.dbm:md5,6000e63ebabe57652cfa8858ef6cc484",
                        "trans.blast.dat.gz:md5,85ba5ea96c566f751ad83a3e4b8ab128"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-10T22:11:39.460127957"
    },
    "STARFUSION_BUILD - human - minigenome - stub": {
        "content": [
            {
                "0": [
                    [
                        "AnnotFilterRule.pm:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "PFAM.domtblout.dat.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "__chkpts:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "blast_pairs.dat.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "blast_pairs.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "fusion_annot_lib.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "fusion_annot_lib.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "pfam_domains.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.cdna.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.cdna.fa.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.cds:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.cdsplus.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.cdsplus.fa.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.gtf:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.gtf.gene_spans:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.gtf.mini.sortu:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.gtf.mm2.splice.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.pep:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.prot_info.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.mm2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.ndb:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.nhr:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.nin:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.njs:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.not:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.nsq:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.ntf:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.nto:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.star.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "trans.blast.align_coords.align_coords.dat:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "trans.blast.align_coords.align_coords.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "trans.blast.dat.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "reference": [
                    [
                        "AnnotFilterRule.pm:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "PFAM.domtblout.dat.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "__chkpts:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "blast_pairs.dat.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "blast_pairs.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "fusion_annot_lib.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "fusion_annot_lib.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "pfam_domains.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.cdna.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.cdna.fa.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.cds:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.cdsplus.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.cdsplus.fa.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.gtf:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.gtf.gene_spans:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.gtf.mini.sortu:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.gtf.mm2.splice.bed:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.pep:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_annot.prot_info.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.fai:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.mm2:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.ndb:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.nhr:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.nin:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.njs:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.not:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.nsq:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.ntf:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.nto:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "ref_genome.fa.star.idx:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "trans.blast.align_coords.align_coords.dat:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "trans.blast.align_coords.align_coords.dbm:md5,d41d8cd98f00b204e9800998ecf8427e",
                        "trans.blast.dat.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-10T22:12:14.50896742"
    }
}